### PR TITLE
Relax PHP dependency so it works on PHP 8.1.

### PR DESCRIPTION
**Pull request by drupol**
Hey hi there,

I would like to use [marcosh/lamphpda](https://github.com/marcosh/lamphpda/) in a PHP 8.1 project and I'm unable to do it because of this requirement package in it.

```
Loading composer repositories with package information
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - roave/you-are-using-it-wrong[1.0.0, ..., 1.4.0] require php ^7.3.0 -> your php version (8.1.6) does not satisfy that requirement.
    - roave/you-are-using-it-wrong[1.5.0, ..., 1.7.0] require php ^7.4.7 -> your php version (8.1.6) does not satisfy that requirement.
    - roave/you-are-using-it-wrong 1.8.0 requires php ^7.4.7 || ~8.0.0 -> your php version (8.1.6) does not satisfy that requirement.
    - roave/you-are-using-it-wrong[1.9.0, ..., 1.10.0] require php ~8.0.0 -> your php version (8.1.6) does not satisfy that requirement.
    - Root composer.json requires roave/you-are-using-it-wrong ^1 -> satisfiable by roave/you-are-using-it-wrong[1.0.0, ..., 1.10.0].
```

This PR:
* Relax `php` dependency so it can be installed with PHP 8.1
* Add `config.allow-plugins` properties in `composer.json`.

Questions:
* Why keeping a `composer.lock` in the repo?
* Should we update `composer.lock` file? If yes, with wich version of PHP?

Thanks in advance!

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
         }
     ],
     "require": {
-        "php":                       "~8.0.0",
+        "php":                       "~8.0.0 || ~8.1.0",
```
Comment (Ocramius) [inline composer.json:13]: 8.1 needs to be checked in CI too

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (drupol) [reply composer.json:13]: Pushed the changes, hope it's ok.

Comment (Ocramius) [reply composer.json:13]: See CI output 😁

Comment (drupol) [reply composer.json:13]: Let me do some meditation first :pray:

**Comment by drupol** [comment]
@Ocramius Could you enable the permission to enable workflow? For some reason it doesn't starts anymore.

**Comment by Ocramius** [comment]
It only starts automatically after the first merged contribution

**Comment by Ocramius** [comment]
Handled as part of #132

**Pull request closed without merging**